lib/arrow.cpp

```diff
@@ -2036,9 +2037,8 @@ namespace grnarrow {
     void
     add_metadata(const char *key, const char *value)
     {
-      arrow::KeyValueMetadata metadata;
-      metadata.Append(key, value);
-      auto status = schema_builder_.AddMetadata(metadata);
+      metadata_.Append(key, value);
+      auto status = schema_builder_.AddMetadata(metadata_);
```

Comment:
If we call this multiple times, are the same metadata added multiple times?

```cpp
writer.add_metadata("x", "1"); // add x=1
writer.add_metadata("y", "2"); // add x=1 and y=2?
```

Reply:
`AddMetadata(metadata_)` is `Set` behavior rather than `Add`.

```c++
writer.add_metadata("x", "1");
writer.add_metadata("y", "2");
```

In that example, x=1 and y=2 are set in the metadata.
There is no such thing as two x=1's being set.

Reply:
Oh.

https://github.com/apache/arrow/issues/46146

Reply:
Could you add a comment that this is "set" not "add"?